### PR TITLE
[MDS-5019] SSO implementation docs

### PR DESCRIPTION
**Pull request by vyasworks**
## Objective 

[MDS-5019](https://bcmines.atlassian.net/browse/MDS-5019)

### CODE REVIEW

File: services/core-api/README.md [hunk 1]
```diff
@@ -91,23 +97,25 @@ under the `__init__.py` file.
...
-### Click commands workflow to transfer files to the object store 
-*Pre-requisite to running the transfer-files, verify-files, and reorganize-files commands: ensure that a Celery worker is running. You can use the command `ps aux` to check.
+### Click commands workflow to transfer files to the object store
+
+\*Pre-requisite to running the transfer-files, verify-files, and reorganize-files commands: ensure that a Celery worker is running. You can use the command `ps aux` to check.
 
 **I want to transfer all documents that exist on the filesystem to the object store**
- 1. Get a list of untransferred files:<br/>
+
+1.  Get a list of untransferred files:<br/>
```
Comment (taraepp) [inline services/core-api/README.md]: I know you didn't add them, but do we need those `<br/>` tags in here?

File: services/core-api/README.md [hunk 2]
```diff
@@ -117,6 +125,86 @@ results. The testing structure is based around [flask testing
...
+
+MDS was originally working with custom keycloack instance (hosted on openshift?) and a local keycloack container for development. This was hard to work with locally and maintain (patching, upgrades etc..)
+
+V2:
+
+The project then moved on to shared keycloack instance (silver SSO) hosted by the platform team and the shared instance had multiple realms provisioned for each ministry division / program.
+
+Benefits:
+
+- Not having to maintain the keycloack instance (the plfrom team does it for us)
```
Comment (taraepp) [inline services/core-api/README.md]: typo: plfrom --> platform

File: services/core-api/README.md [hunk 3]
```diff
@@ -117,6 +125,86 @@ results. The testing structure is based around [flask testing
...
+
+There were several issues with this approach for the platform team (it was great for teams that had full realms though!)
+
+Cons:
+
+- Having a large number of realms on keycloak created severe performance penalty as the number of projects kept getting higher.
+- Most teams required the same features (IDIR, BCeID login) but had different implementations.
+- Some teams required specalized features that needed instance level changes (hard coupling)
+
+Since most projects need the standard setup anyways, it was deicede to move to an offering that does the basic things very well and specialized requirements will be handled by the teams themselves, with custom keycloak implementations.
```
Comment (taraepp) [inline services/core-api/README.md]: typo: deicede --> decided

**Review by taraepp — CHANGES_REQUESTED** [review]
Couple of typos but I like the explanation of the history, that's really helpful in understanding things.

File: services/core-api/README.md (hunk 2 above)
Comment (vyasworks) [reply services/core-api/README.md]: ew, how did this happen..

File: services/core-api/README.md (hunk 1 above)
Comment (vyasworks) [reply services/core-api/README.md]: strange, wondering what added this tag over here 🤔

**Review by taraepp — APPROVED** [review]
👍

**Pull request merged**